src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,104 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList() {
+        this.elementData = new Object[DEFAULT_CAPACITY];
+    }
+
+    public void ensureCapacity() {
+        Object[] newElementData = Arrays.copyOf(elementData, size + 1);
```

Comment:
new size should be 1,5 times bigger as previous. you should resize your array only when it's full